flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -254,55 +272,89 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
   }
 
   Future<void> _finishTimeToDisplayTracking() async {
-    _timeToDisplayTracker?.clear();
+    try {
+      final transaction = _transaction;
+      _transaction = null;
+      if (transaction == null || transaction.finished) {
+        return;
+      }
 
-    final transaction = _transaction;
-    _transaction = null;
-    if (transaction == null || transaction.finished) {
-      return;
+      // Cancel unfinished child spans, e.g this might happen if the user navigates
+      // away from the current route before TTFD or TTID is finished.
+      for (final child in (transaction as SentryTracer).children) {
```

Comment:
we can't just finish all spans of the transaction, as we would finish the spans started by the user, too.
we should just finish TTID and TTFD (with status deadline_exceeded).
Also, we can remove the transaction from the scope and let it finish in the background. wdyt?

Reply:
> we should just finish TTID and TTFD (with status deadline_exceeded)

👍 

> we can remove the transaction from the scope and let it finish in the background. wdyt?

I think that's fine, probably is better than forcefully finishing it

Reply:
what if we remove it from the scope but still finish it without awaiting?

Reply:
> what if we remove it from the scope and still finish it without awaiting?

yep, that's the idea